amethyst_assets/src/asset.rs

```diff
@@ -23,6 +23,15 @@ pub trait Asset
 where
     Self: Clone + Sized,
 {
+    /// The `Context` type that can produce this asset
+    type Context: Context<Asset=Self, Data=Self::Data, Error=Self::Error>;
+
+    /// The `Data` type the asset can be created from.
+    type Data;
```

Comment:
They are just like aliases to write `A::Error` instead of `<A::Context as Context::Error`.